maciekmm/origami
# Community

What's left to do:
- [x] Fix CI for solver
- [x] Add CI for community backend
- [x] Move requirements.txt to /backend
- [x] Hide secret / mail server configuration/media_root behind environment variables
- [x] Add required libraries to setup.py (solver)

How to run/test:
- Make sure to have virtual env in /backend dir
- Install requirements from both *solver* and *community*
- Run `python setup.py install` inside solver directory
- Run redis `docker run --name origuide-redis -p '127.0.0.1:6379:6379' -d redis`
- Run celery worker `celery -A community worker --loglevel=info`
- Run migrations using `python.py manage.py migrate`
- Run mock SMTP server `python -m smtpd -n -c DebuggingServer localhost:1025`
- Run backend using `python.py manage.py runserver` **or** run tests via the `test` action in manage.py
- You can use the following insomnia requests [insomnia.tar.gz](https://github.com/maciekmm/origami/files/5287323/insomnia.tar.gz) to aid testing.
